qt/cmake-project/cmake-project

```diff
@@ -75,21 +92,66 @@ fi
 if [ "$QTVERSION" != "none" ]; then
 cat >> CMakeLists.txt <<EOF
 find_package($QT CONFIG REQUIRED COMPONENTS $QTLIBS)
+
+include(cmake/osdeployqt.cmake)
+
 set(CMAKE_AUTOMOC TRUE)
 set(CMAKE_AUTORCC TRUE)
-$AUTOUIC
-
+set(CMAKE_AUTOUIC TRUE)
+$AUTOUIC	
 EOF
 fi
 
 cat >> CMakeLists.txt <<EOF
-add_executable($PROJECTNAME
-$FILES
+set(SRC
+$SRC
+)
+
+EOF
+cat >> CMakeLists.txt <<EOF
+set(HEADERS
+$HEADERS
+)
+
+EOF
+cat >> CMakeLists.txt <<EOF
+set(QRCS
+$QRCS
+)
+
+EOF
+cat >> CMakeLists.txt <<EOF
+set(UI_FOLDERS
+$UI_FOLDERS
+)
+
+set(CMAKE_AUTOUIC_SEARCH_PATHS
+\${UI_FOLDERS}
+)
+
+set(UIS
+$UIS
+)
+
+EOF
+
+cat >> CMakeLists.txt <<EOF
+set(INCLUDE_FOLDERS
+	$INCLUDES
 )
+
+include_directories(
+	\${INCLUDE_FOLDERS}
+)
+
+add_executable($PROJECTNAME \${SRC} \${HEADERS} \${QRCS} \${UIS})
```

Comment:
We might as well use modern cmake and use target_sources() instead of all those variables.

(target_sources exists since cmake 3.1, so it should be fine for everyone)

Reply:
I am not that advanced in CMake but you could always change my code to the best practice.